Source/Evolve/WarpXEvolve.cpp

```diff
@@ -79,39 +79,36 @@ namespace
 }
 
 void
-WarpX::Synchronize () {
+WarpX::SynchronizeVelocityWithPosition () {
     using ablastr::fields::Direction;
     using warpx::fields::FieldType;
 
-    // Note that this is potentially buggy since the PushP will do a field gather
-    // using particles that have been pushed but not yet checked at the boundaries.
-    // Also, this PushP may be inconsistent with the PushP backwards above since
-    // the fields may change between the two (mainly effecting the Python version when
-    // using electrostatics).
-    // When synchronize_velocity_for_diagnostics is true, the PushP at the end of the
-    // step is used so that the correct behavior is obtained.
-    FillBoundaryE(guard_cells.ng_FieldGather);
-    FillBoundaryB(guard_cells.ng_FieldGather);
-    if (fft_do_time_averaging)
-    {
-        FillBoundaryE_avg(guard_cells.ng_FieldGather);
-        FillBoundaryB_avg(guard_cells.ng_FieldGather);
-    }
-    UpdateAuxilaryData();
-    FillBoundaryAux(guard_cells.ng_UpdateAux);
-    for (int lev = 0; lev <= finest_level; ++lev) {
-        mypc->PushP(
-            lev,
-            0.5_rt*dt[lev],
-            *m_fields.get(FieldType::Efield_aux, Direction{0}, lev),
-            *m_fields.get(FieldType::Efield_aux, Direction{1}, lev),
-            *m_fields.get(FieldType::Efield_aux, Direction{2}, lev),
-            *m_fields.get(FieldType::Bfield_aux, Direction{0}, lev),
-            *m_fields.get(FieldType::Bfield_aux, Direction{1}, lev),
-            *m_fields.get(FieldType::Bfield_aux, Direction{2}, lev)
-        );
+    if (!is_synchronized) {
```

Comment:
This may be one of those examples where making it clear that `is_synchronized` is a member variable of the `WarpX` class, through the prefix `m_`, would help not get lost while reading the code. For a minute I was trying to understand where `is_synchronized` is defined in this function. We could rename it according to the WarpX style rules in a separate PR.